packages/itwinui-css/src/table/base.scss

```diff
@@ -39,6 +39,7 @@
   flex-shrink: 0;
   font-weight: var(--iui-font-weight-semibold);
   border-block-end: 1px solid var(--iui-color-border);
+  border-radius: inherit;
```

Comment:
Adding `border-radius` to `.iui-table` definitely seems to be the way to go, since the `.iui-table-header-wrapper` class doesn't have the full width of the table anymore.
The `.iui-table-header-wrapper` in iTwinUI v2:
![image](https://github.com/user-attachments/assets/3e5d8718-d9ca-4695-8dfe-4458fa128231)
The `.iui-table-header-wrapper` in iTwinUI v3:
![image](https://github.com/user-attachments/assets/79112e17-0465-44f1-8e8e-87ba01795ec3)
The scrollbar seems to look okay with the curved corners on Firefox and Chrome.
Fireforx:
![image](https://github.com/user-attachments/assets/bc501c3e-25e0-45e8-bbae-a40d394c24a1)
Chrome:
![image](https://github.com/user-attachments/assets/4f04a863-60b8-4b03-9d4c-15040058517c)

Border radius inheritance is a trickier issue here, since we do draw borders around the expandable sub content on `Table`s, and if those borders don't also have rounded corners, then they will slightly clip out of `Table` See the bottom left corner of this screenshot: 

![image](https://github.com/user-attachments/assets/4abb4b17-0c34-4a98-a421-f7f749feafc3)

It probably makes more sense to define the border radius on that corner directly, instead of relying on a chain of inheritance like we currently do, since that chain can be broken. It might make sense for the bottom right corner of this expandable content to have a square corner, though, since it isn't directly on the rounded corner of the `Table` like the bottom left corner is. What do you think? Should the bottom right corner of the expandable content be squared, or rounded?

Reply:
I've added `border-radius` to the `.iui-table` class in this [commit](https://github.com/iTwin/iTwinUI/pull/2143/commits/8c863354915af873335a9d64779a851867aa3e19). Also updated changesets and the PR description to reflect this change.